src/Form/QuestionType/QuestionTypeCategory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Form\QuestionType;
+
+/**
+ * List of valid question types categories
+ */
+enum QuestionTypesCategory: string
```

Comment:
Please make sure that using an enum will not prevent us to add the ability to register more types from plugins.

Reply:
I've given it some thought last week and it seems ok to me.

Enum are allowed to implemeted interfaces, which mean we can have a common `QuestionTypesCategory` interface which is them implemented by a `CoreQuestionTypesCategory` enum and any number of plugin supplied classes.

I guess we can already rename the enum to `CoreQuestionTypesCategory` and create the separate `QuestionTypesCategory` interface to be ready for this ?

Reply:
If you consider it will not be an issue, there is no need to rename it now. It could be done in the PR that will give the ability to the plugins to declare their own question types.

Reply:
👍